File: go/WhileLoop.go
// Go program to illustrate the
// the for loop as while Loop
package main

import "fmt"

// Main function
func main() {

	// while loop for loop executes till  i < 3 condition is true, change i to -1 to show a bad loop
	i := 0
	for i < 5 {
		i += 1
		fmt.Println("While loop implementation in Golang ", i)
	}
	fmt.Println("Done")
}

File: c/WhileLoop.c

// C program to illustrate while loop
#include <stdio.h>
  
int main()
{
    // initialization expression
    int i = 1;
  
    // test expression
    while (i < 5)
    {
        printf( "Welcome to while loop in C %d\n",i);    
  
        // update expression
        i++;
        printf( "i is now %d\n",i);
    }
  
    return 0;
}